.github/workflows/main.yml

```diff
@@ -58,3 +65,33 @@ jobs:
         path: |
           _build/test/logs/
           !_build/test/logs/last
+
+  slack:
+    needs: test
+    runs-on: ubuntu-20.04
+    if: always()
+    steps:
+    - name: Slack notification
+      uses: 8398a7/action-slack@v3
+      with:
+        author_name: "GitHub Actions"
+        username: ${{ github.event.repository.name }}
+        icon_emoji: ':octocat:'
+        fields: repo, message, ref, commit, author, action, eventName, workflow, job, took
+        status: ${{ needs.test.result }}
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK }}
+        MATRIX_CONTEXT: ${{ toJson(matrix) }}
+
+  finish:
+    needs: test
+    runs-on: ubuntu-20.04
```

Comment:
finish is not called when on the test fails, there should be a `if: always()` here.

Reply:
Yes, but I was think that the testes must pass before call finish. Can you confirm that this behavior(`if: always()`) is also expected for other repo?

Reply:
yes, I think so. Otherwise we are not going to get the coverage for failed tests.

Reply:
Thanks. Done.